modP2pImpl/src/org/aion/p2p/impl1/tasks/ChannelBuffer.java

```diff
@@ -123,16 +120,23 @@ synchronized boolean shouldRoute(int _route, int _maxReqsPerSec) {
         }
     }
 
-
     RouteStatus getRouteCount(int _route) {
         return routes.get(_route);
     }
 
     void readHead(ByteBuffer buf) {
+        if (buf.array().length < bsHead.length) {
+            if (p2pLOG.isDebugEnabled()) {
+                p2pLOG
+                    .debug("ChannelBuffer readHead short buffer size");
+            }
+            return;
+        }
+
         buf.get(bsHead);
         try {
             header = Header.decode(bsHead);
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
What exceptions is this supposed to catch?  Header#decode has no declared Exceptions.  It can throw IllegalArgumentException or IndexOutOfBoundsException, but to catch that you'll have to catch those specifically, or catch RuntimeException